### PR TITLE
test cypress 12 upgrade

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -385,16 +385,18 @@ jobs:
     working_directory: ~/
     steps:
       - run:
+          # TODO: upgrade image with newer Node.js and remove --ignore-engines
```
Comment (svidgen) [inline .circleci/config.yml:388]: Does this do the trick? https://github.com/aws-amplify/amplify-js/pull/10876

Comment (jimblanc) [reply .circleci/config.yml:388]: It should! We're going to get the upgrade merged first, and once it's out I can update my PR to remove the flag.

Comment (david-mcafee) [reply .circleci/config.yml:388]: @svidgen prioritizing getting this in the pipeline, but yes, will revisit after @jimblanc's PR is merged :)

File: .circleci/config.yml [hunk 2]
```diff
@@ -385,16 +385,18 @@ jobs:
...
           name: 'Clone Amplify JS Samples repo, install cypress, install react samples authenticator and link amplify packages'
           command: |
             mkdir ~/.ssh/ && echo -e "Host github.com\n\tStrictHostKeyChecking no\n" > ~/.ssh/config
             echo $SSH_HOST_PUBLIC_KEY >> ~/.ssh/known_hosts
             git clone $AMPLIFY_JS_SAMPLES_STAGING_URL
             cd amplify-js-samples-staging
+            git checkout cypress-12-upgrade
```
[the remark this answers is not in the document]
Comment (david-mcafee) [reply .circleci/config.yml:395]: Thanks @jimblanc, I wasn't aware of that! That's a fantastic update!

File: .circleci/config.yml [hunk 3]
```diff
@@ -1427,7 +1429,7 @@ jobs:
             git config --global user.name $GITHUB_USER
             git status
             git --no-pager diff
-            yarn publish:$CIRCLE_BRANCH
+            # yarn publish:$CIRCLE_BRANCH
```
Comment (david-mcafee) [inline .circleci/config.yml:1432]: Will remove prior to merge

File: .circleci/config.yml [hunk 4]
```diff
@@ -1465,6 +1466,7 @@ releasable_branches: &releasable_branches
       - 1.0-stable
       - geo/main
       - in-app-messaging/main
+      - cypress-12-upgrade
```
Comment (david-mcafee) [inline .circleci/config.yml:1469]: Will remove prior to merge

**Review by svidgen — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**